airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryUtils.java

```diff
@@ -115,6 +116,20 @@ public static void createDataset(final BigQuery bigquery, final String datasetId
     }
   }
 
+  public static void checkHasCreateAndDeleteDatasetRole(final BigQuery bigquery, final String datasetId, final String datasetLocation) {
```

Comment:
There has been a push to make error messages more descriptive and actionable. I have a PR out here https://github.com/airbytehq/airbyte/pull/18560 that will change handling config errors. TLDR once that PR is in the change to made is:
- throw a ConfigErrorException with the message you'd like to show to the user if this check fails

If this is more urgent and can't wait for that PR, then at the very least populate the message assc with the ``AirbyteConnectionStatus`` with a relevant message.

Reply:
Hi @akashkulk. It's ok, I think I may wait. Can I ask you to ping me please when you're done with your changes?

Reply:
@etsybaev I have merged PR: https://github.com/airbytehq/airbyte/pull/18989 which should unblock you. You can throw a ConfigErrorException here

Reply:
Updated this PR, thanks